controllers/clustertopology_controller_sync.go

```diff
@@ -99,3 +108,193 @@ func checkReferencedObjectsAreStrictlyCompatible(current, desired client.Object)
 	}
 	return checkReferencedObjectsAreCompatible(current, desired)
 }
+
+// reconcileMachineDeployments reconciles the desired state of the MachineDeployment objects.
+func (r *ClusterTopologyReconciler) reconcileMachineDeployments(ctx context.Context, current, desired *clusterTopologyState) error {
+	currentMachineDeployments, err := getMachineDeploymentsByID(current.machineDeployments)
+	if err != nil {
+		return err
+	}
+	desiredMachineDeployments, err := getMachineDeploymentsByID(desired.machineDeployments)
+	if err != nil {
+		return err
+	}
```

Comment:
nit: I will move this into calculateMachineDeploymentDiff, given that those variable are used only there

Reply:
I did that, but then I saw that I also need them below in the for loops. I could move them but then I have to change the returned data structures. It's relatively simple for create/delete but for update I would have to return a tuple of current/desired MD.

Reply:
I'll resolve it as it's solved by using a map in clusterTopologyState